lib/dealership.rb

```diff
@@ -0,0 +1,57 @@
+class Dealership
+  attr_reader :name,
+              :address,
+              :inventory,
+              :inventory_count
+
+  def initialize(name, address)
+    @name            = name
+    @address         = address
+    @inventory       = []
+    @inventory_count = 0
+  end
+
+  def add_car(car)
+    @inventory << car
+    @inventory_count += 1
+  end
+
+  def has_inventory?
+    @inventory_count > 0
+  end
+
+  def cars_by_make(make)
+    @inventory.select do |vehicle|
+      vehicle.make == make
+    end
+  end
+
+  def total_value
+    @inventory.sum do |vehicle|
+      vehicle.total_cost
+    end
+  end
+
+  def details
+    {
+    "total_value" => total_value,
+    "address" => @address
+    }
+  end
+
+  def average_price_of_car
+    (total_value / @inventory_count).to_s.insert(2, ",")
```

Comment:
This works for when we need one comma in a 5 digit number, but what would we do if we were working in the millions or billions?

Here is some further reading of how we could handle that https://rick-moore.medium.com/formatting-number-strings-in-ruby-4da35d5282e3